modules/ugdk-core/include/ugdk/graphic/mesh.h

```diff
@@ -0,0 +1,91 @@
+
+#ifndef UGDK_GRAPHIC_MESH_H_
+#define UGDK_GRAPHIC_MESH_H_
+
+#include <ugdk/structure/vertex.h>
+#include <ugdk/graphic/canvas.h>
+#include <ugdk/graphic/module.h>
+#include <ugdk/graphic/vertexdata.h>
+#include <ugdk/graphic.h>
+
+#include <glm/vec2.hpp>
+
+#include <vector>
+
+namespace ugdk {
+namespace graphic {
+
+class Canvas;
+
+// TODO:
+// + [x] Add fill methods
+// + [x] Write stream method
+// Remember:
+// + VertexData will be used internally to group buffers
+// + Mesh is used by user for easy access
+
+template <typename Vertex>
+class Mesh {
+  public:
+    /// Alias
+    using VertexArray = std::vector<Vertex>;
+
+    /// Default constructor (an invalid mesh)
+    Mesh();
+
+    /// Construct mesh with given draw mode
+    Mesh(const DrawMode& the_mode, GLTexture* the_texture);
+
+    /// 
+    void Fill(const VertexArray& vertices);
+
+    void StreamTo(Canvas &canvas) const;
+    
+  private:
+    DrawMode            mode_;
+    GLTexture           *texture_;
+    std::vector<Vertex> vertices_;
+};
+
+using Mesh2D = Mesh<structure::VertexXYUV>;
+
+template <typename Vertex>
+inline Mesh<Vertex>::Mesh() : Mesh(DrawMode::TRIANGLES(), nullptr) {}
+
+template <typename Vertex>
+inline Mesh<Vertex>::Mesh(const DrawMode& the_mode, GLTexture* the_texture)
+    : mode_(the_mode), texture_(the_texture) {}
+
+template <typename Vertex>
+inline void Mesh<Vertex>::Fill(const Mesh<Vertex>::VertexArray& vertices) {
+    vertices_ = vertices;
+}
+
+template <typename Vertex>
+Canvas& operator<<(Canvas &canvas, const Mesh<Vertex> &mesh) {
+    mesh.StreamTo(canvas);
+    return canvas;
+}
+
+template <typename Vertex>
+void Mesh<Vertex>::StreamTo(Canvas &canvas) const {}
```

Comment:
This should be something that results in a compilation error if used, not an empty function.

Reply:
I did try using static assertion, but it gave a compile error, as if it always compiled through the assertion. Never did one before, will try again.

Reply:
It seems this strategy solves the problem:
https://stackoverflow.com/questions/8854286/how-to-enforce-use-of-template-specialization